### ISSUE TITLE
Some changes in the Portuguese Corpora

### ISSUE THREAD

**Issue by limapedro** [common-voice/common-voice#2537]
Hi, I've seen some errors in the Portuguese corpora, some phrases lack a period, this one seems easy to fix, some offensive phrases that I've tagged a while ago showed up in the dataset that was released last week. I've made a quick fix that could be commited right away.
Here's a link to the file. 
https://drive.google.com/file/d/1vp1q0utgb8FMrpO8kfDKNhR-j3_TgChz/view?usp=sharing

**Comment by phirework** [comment]
Hi, can you clarify what the file links to? Is it the corrected set of the Portuguese sentences? If so, you will also need to open a ticket in https://github.com/Common-Voice/sentence-collector/, otherwise the removed sentences will be just re-added next time there is an export. 

If someone has recorded a clip with a mistake in it and that sentence then gets flagged, we currently have no process for retroactively removing it from the compiled corpora. Thanks for bringing this to our attention, I will add this to our roadmap.

**Comment by limapedro** [comment]
@phirework  I'm working on correcting grammar errors from the Pt repo, I'm afraid that the next batch of sentences from the collection tool will come with a few errors again, about the file is pt/sentence-collector.txt, also improving the quality of the sentences is my top priority now, the collection tool has a flaw, these sentences sholdn't have reappeared in the corpora, thanks for the reply. I'll sent updates later.

**Comment by Felizolinha** [comment]
The portuguese corpora definitely needs some fixing. One thing that I believe is still broken in your fixed version, is that many sentences have random question marks in the middle of the phrases. Loads of phrases are also missing a capital letter after the punctuation.

**Comment by limapedro** [comment]
@Felizolinha Yeah I've notice this, I'm still working on correcting these mistakes and  improving readability of the sentences. When I'm done I'll ask for a commit. For the time being I think is better to stop the incoming sentences from the collector until we have ways of correcting them at scale.

**Comment by MichaelKohler** [comment]
As @phirework said we will need to delete wrong sentences in the Sentence Collector as well. For that I'll need a list of the sentences to delete, preferably in a text file with one sentence per list.

I'd also like to note that there is no Portuguese validation file for the Sentence Collector, adding one would IMHO be way better than always fixing after an export. You can find more information here: https://github.com/Common-Voice/sentence-collector/blob/master/shared/validation/VALIDATION.md . You can copy the English one as base (that's what is currently being used as fallback). I'm happy to help out there if you need any help.

Can I also have a few examples of what is wrong with the current sentences?

> For the time being I think is better to stop the incoming sentences from the collector until we have ways of correcting them at scale.

I'm not sure if I like this. In theory we could do that, and we just don't export pt for now. However I fear this would also hold back many valid sentences.

**Comment by limapedro** [comment]
@MichaelKohler Hi, I'm working on fixing the sentences on the current repo, so many sentences lack a period, some sentences have a question mark out of nowhere, despite that the rest of the sentences are good, need some improving, about your last comment now we have some ground truth for valid and invalid sentences, I'm pretty we could come up with some solution to this problem, I'll update you as soon as possible.

**Comment by MichaelKohler** [comment]
I'm not sure if there is much value in changes all sentences to have a period at the end. Some consistency is good, but in this case we assume each sentence is actually a full sentence.

**Comment by limapedro** [comment]
All sentence that I've seen so far are full sentences. This is an update to sentences for the Portuguese Corpora, I've removed some sentences, just a few, changed for more consistency and fixed the errors mentioned above. https://drive.google.com/file/d/1bpytjhUlL17-H-q9-LNk--Oc8J3Hyehr/view?usp=sharing 

Could you get this up and running?

**Comment by Felizolinha** [comment]
> I'm not sure if there is much value in changes all sentences to have a period at the end. Some consistency is good, but in this case we assume each sentence is actually a full sentence.

What if we enforce punctuation only for the newly added sentences, by using a validator?

**Comment by MichaelKohler** [comment]
> What if we enforce punctuation only for the newly added sentences, by using a validator?

That could be an option, however I'd first like to know from the Common Voice team if that adds any value. @nukeador what do you think?

**Comment by nukeador** [comment]
My understanding is that deep speech removes periods. @MichaelKohler we should check with Reuben or Kelly.

**Comment by nukeador** [comment]
Also I suggest we move this conversation to discourse, and keep github just for changes we have agreed.

**Comment by MichaelKohler** [comment]
> My understanding is that deep speech removes periods. @MichaelKohler we should check with Reuben or Kelly.

That's correct. From a DeepSpeech perspective it doesn't matter. It would only matter in terms of having a consistent data set.

**Comment by limapedro** [comment]
@MichaelKohler @nukeador will you guys change the current repo? Since the quality of the sentences is better and without many mistakes. We could focus on word distribuition, since Portuguese is somewhat phonetically consistent I think is a good idea to focus on covering as many words as possible with less sentences, have anyone thought of this before?

**Comment by ftyers** [comment]
Closing this one as stale.
